File: src/ircbot/BotStates.java
/*
 *  UniBot
 *  Copyright (C) 2009 Florian Ziesche
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>
 */

package ircbot;

import java.lang.String;
import java.util.ArrayList;
import java.util.Collection;

/**
 *
 * @author flo
 */
public class BotStates {
	private Logger log;
	private Config config;

	// bot states:
	private boolean connected;
	private boolean joined;
	private boolean op;
	private boolean quit;
	private boolean identified;
	private ArrayList<String> userList;

	public BotStates(Logger log, Config config) {
		this.log = log;
		this.config = config;

		userList = new ArrayList<String>();

		connected = false;
		joined = false;
		op = false;
		quit = false;
		identified = false;
	}

	public boolean isConnected() {
		return connected;
	}

	public void setConnected(boolean connected) {
		this.connected = connected;
	}

	public boolean isJoined() {
		return joined;
	}

	public void setJoined(boolean joined) {
		this.joined = joined;
	}

	public boolean isOp() {
		return op;
	}

	public void setOp(boolean op) {
		this.op = op;
	}

	public boolean isQuit() {
		return quit;
	}

	public void setQuit(boolean quit) {
		this.quit = quit;
	}

	public boolean isIdentified() {
		return identified;
	}

	public void setIdentified(boolean identified) {
		this.identified = identified;
	}

	public void addUser(String name) {
		userList.add(name);
	}

	public void deleteUser(String name) {
		if(userList.contains(name)) {
			userList.remove(name);
		}
	}

	public void updateUser(String from, String to) {
		if(userList.contains(from)) {
			userList.set(userList.indexOf(from), to);
		}
	}

	public void deleteAllUsers() {
		userList.clear();
	}

	public String[] getUsers() {
		return (String[])userList.toArray(new String[0]);
	}

}

File: bot_states.cpp
/*
 *  UniBot
 *  Copyright (C) 2009 Florian Ziesche
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General License for more details.
 *
 *  You should have received a copy of the GNU General License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>
 */

#include "bot_states.h"

bot_states::bot_states() {	
	connected = false;
	joined = false;
	parted = false;
	op = false;
	quit = false;
	kicked = false;
	identified = false;
	kick_user = "";
}

bot_states::~bot_states() {
}

bool bot_states::is_connected() {
	return connected;
}

void bot_states::set_connected(bool connected) {
	this->connected = connected;
}

bool bot_states::is_joined() {
	return joined;
}

void bot_states::set_joined(bool joined) {
	this->joined = joined;
}

bool bot_states::is_parted() {
	if(parted) {
		parted = false;
		return true;
	}
	return false;
}

void bot_states::set_parted(bool parted) {
	this->parted = parted;
}

bool bot_states::is_kicked() {
	if(kicked) {
		kicked = false;
		return true;
	}
	return false;
}

void bot_states::set_kicked(bool kicked) {
	this->kicked = kicked;
}

bool bot_states::is_op() {
	return op;
}

void bot_states::set_op(bool op) {
	this->op = op;
}

bool bot_states::is_quit() {
	return quit;
}

void bot_states::set_quit(bool quit) {
	this->quit = quit;
}

bool bot_states::is_identified() {
	return identified;
}

void bot_states::set_identified(bool identified) {
	this->identified = identified;
}

void bot_states::add_user(string name, string real_name, string host) {
	user_list[name] = pair<string, string>(real_name, host);
}

void bot_states::update_user(string name, string real_name, string host) {
	user_list[name] = pair<string, string>(real_name, host);
}

void bot_states::delete_user(string name) {
	if(user_list.count(name) > 0) {
		user_list.erase(name);
	}
}

void bot_states::update_user(string from, string to) {
	if(user_list.count(from) > 0) {
		user_list[to] = user_list[from];
		user_list.erase(user_list.find(from));
	}
}

void bot_states::delete_all_users() {
	user_list.clear();
}

map<string, pair<string, string> >* bot_states::get_users() {
	return &user_list;
}

string bot_states::get_kick_user() {
	return kick_user;
}

void bot_states::set_kick_user(string kick_user) {
	this->kick_user = kick_user;
}

pair<string, string> bot_states::get_user(string name) {
	if(user_list.count(name) == 0) return pair<string, string>("", "");
	return user_list[name];
}

bool bot_states::is_user(string name) {
	return (user_list.count(name) > 0);
}
